src/datagrid/DataGrid.tsx

```diff
@@ -201,21 +201,20 @@ const useAutoUpdateRecords = ({
      * Making updatedRowIndex return tuple ensures that the useEffect will run always
      */
     const updatedRowIndex = useMemo((): [number] => {
-        if (hasRowSizeUpdated) return [0];
         if (records === oldRecords.current) return defaultEmptyTuple;
 
         return [records.findIndex((record, index) => oldRecords.current[index] !== record)];
-    }, [oldRecords, records, hasRowSizeUpdated, defaultEmptyTuple]);
+    }, [oldRecords, records, defaultEmptyTuple]);
 
     /**
      *
      * Reset the row height cache if `updatedRowIndex` or `rowSize` changes
      *
      */
     useEffect(() => {
-        if (updatedRowIndex.at(0) === -1) return;
+        if (prevRowSize.current === rowSize && updatedRowIndex.at(0) === -1) return;
```

Comment:
should the condition here be `prevRowSize.current === rowSize || updatedRowIndex.at(0) === -1`